# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBObjectStore.cpp
# Commit: 23fc7469065d
# Full Hash: 23fc7469065d32effcad818083aa3840c95f26d5
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-11 21:53:06
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Reduce code duplication for different open cursor variants. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40956
# ==============================================================================

diff -r 0e77abe58616 -r 23fc7469065d dom/indexedDB/IDBObjectStore.cpp
--- a/dom/indexedDB/IDBObjectStore.cpp	Wed Sep 11 11:51:26 2019 +0000
+++ b/dom/indexedDB/IDBObjectStore.cpp	Wed Sep 11 11:51:24 2019 +0000
@@ -2270,22 +2270,14 @@
 
   IDBCursor::Direction direction = IDBCursor::ConvertDirection(aDirection);
 
-  OpenCursorParams params;
-  if (aKeysOnly) {
-    ObjectStoreOpenKeyCursorParams openParams;
-    openParams.commonParams().objectStoreId() = objectStoreId;
-    openParams.commonParams().optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.commonParams().direction() = direction;
-
-    params = std::move(openParams);
-  } else {
-    ObjectStoreOpenCursorParams openParams;
-    openParams.commonParams().objectStoreId() = objectStoreId;
-    openParams.commonParams().optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.commonParams().direction() = direction;
-
-    params = std::move(openParams);
-  }
+  const CommonOpenCursorParams commonParams = {
+      objectStoreId, std::move(optionalKeyRange), direction};
+
+  // TODO: It would be great if the IPDL generator created a constructor
+  // accepting a CommonOpenCursorParams by value or rvalue reference.
+  const auto params =
+      aKeysOnly ? OpenCursorParams{ObjectStoreOpenKeyCursorParams{commonParams}}
+                : OpenCursorParams{ObjectStoreOpenCursorParams{commonParams}};
 
   RefPtr<IDBRequest> request = GenerateRequest(aCx, this);
   MOZ_ASSERT(request);
